### DISCUSSION TITLE
Regex Attachment Handling - Capture Everything Before First Occurrence

### DISCUSSION THREAD

**Discussion by willjones9981** [nextgenhealthcare/connect#6120] [Q&A] [answered]
I am trying to come up with a regex attachment handler that captures everything prior to my the first occurrence of my delimiter (^^^)

Sample message:

    aaaaa^^^bbbbb^^^123456789^^^end

My current regex:

    [\s\S]*?(?=\^\^\^)

My desired result:

    ${ATTACH65435683}^^^bbbbb^^^123456789^^^end

My current result:

    ${ATTACH:1901033}^${ATTACH:i23048505}^^^123456789^end

So what I have is working in a sense, but it is capturing more than one instance. I just want to capture everything before the first occurrence of "^^^" in an attachment. How can I edit my regex to make this work?

**Comment by pacmano1** [comment, marked as answer]
`(.*?)(?:\^\^\^|$)`

Maybe?

**Reply by willjones9981** [reply]
That is making turning everything between the "^^^" into an attachment. So I end up with ${ATTACH:1901033}^^^${ATTACH:i23048505}^^^${ATTACH:2o203o4d}

**Reply by pacmano1** [reply]
was on my phone, try `^(.*?)(?:\^\^\^|$)`

**Reply by willjones9981** [reply]
nailed it! Thank you so much!
